src/components/Item.jsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import db from '../lib/firebase';
+
+const Item = ({ name, id, lastPurchasedDate }) => {
+  const checkHandler = () => {
+    db.collection('items').doc(id).update({ lastPurchasedDate: new Date() });
+  };
+
+  const checkDate = (lastPurchasedDate) => {
+    if (lastPurchasedDate === null) {
+      return false;
+    } else {
+      const day = 60 * 60 * 24;
+      const now = Date.now() / 1000;
+      return now - lastPurchasedDate.seconds < day;
```

Comment:
When dealing with time, it can get quite tricky. It's a really good idea to be storing the time as a Timestamp in the database like you're doing. When doing any time comparisons, it's best to avoid converting to a number as it can cause some unexpected headaches. Instead, make comparisons on the Date types themselves. Here's an example of that may look:
```js
const purchasedMoreThan24HoursAgo =
      lastPurchasedDate.toDate() <
      new Date(new Date().getTime() - 24 * 60 * 60 * 1000);
```